compiler-core/src/type_.rs

```diff
@@ -55,6 +55,7 @@ pub enum Type {
     ///
     Named {
         publicity: Publicity,
+        from_internal_module: bool,
```

Comment:
I don't think this information should be part of the type itself, rather we can look up whether the module that defines the type is internal.
Generally I'd like to avoid adding anything new to the type data structures.

We'll need to store whether a module is internal for this ticket too https://github.com/gleam-lang/gleam/issues/2891